### PR TITLE
Revise reducer. Removed helper functions.

### PR DESCRIPTION
**Pull request by rrustia**

### CODE REVIEW

File: client/src/reducers/index.js [hunk 1]
```diff
@@ -1,249 +1,255 @@
...
 export const initialState = {
         hasError: false,
         location: "none",
         opening: "0",
         openingValue: "0",
-        board: 
+        board:
             // row 1(top), from left to right
             {
-            a8: {piece: C.bQr, color: "white", row: "1"},
```
Comment (rrustia) [inline client/src/reducers/index.js]: removed unneeded attributes from board.

File: client/src/reducers/index.js [hunk 2]
```diff
@@ -1,249 +1,255 @@
...
-            b1: {piece: C.wQk, color: "white", row: "8"},
-            c1: {piece: C.wQb, color: "black", row: "8"},
-            d1: {piece: C.wQ, color: "white", row: "8"},
-            e1: {piece: C.wK, color: "black", row: "8"},
-            f1: {piece: C.wKb, color: "white", row: "8"},
-            g1: {piece: C.wKk, color: "black", row: "8"},
-            h1: {piece: C.wKr, color: "white", row: "8"}
-        },               
-        // This board would be used if board was inverted.
-        boardInverted: {
```
Comment (rrustia) [inline client/src/reducers/index.js]: Deleted boardInverted for now. Might be implemented in future.

File: client/src/reducers/index.js [hunk 3]
```diff
@@ -1,249 +1,255 @@
...
-            merge.board[location].piece = pickedUpPiece;
-            merge.board[origin].piece = "";
-            merge.carry = false;
-            merge.moveIndex = state.moveIndex + 1;
-            // after making a correct move, let the computer play the next move
-            let nextMoveIndex, mColor;
-            (color === "White") ? (nextMoveIndex = state.moveIndex, mColor = "black") : ( nextMoveIndex = state.moveIndex + 1, mColor = "white");
-            merge.board = move(merge.board, state.opening, nextMoveIndex, mColor );
+    if (action.type === actions.BOARD_LOCATION) {
+        // scenario: board is currently disabled.
```
Comment (visaudi) [inline client/src/reducers/index.js]: Get rid of unnecessary comments.

File: client/src/reducers/index.js [hunk 4]
```diff
@@ -1,249 +1,255 @@
...
+        if (state.boardDisabled) {
+            return Object.assign({}, state, {
+                feedbackMessage: "Press Start to begin."
+            });
+        };
+        const color = state.userColor;
+        const move = action.move;
+        let origin, destination;
```
Comment (visaudi) [inline client/src/reducers/index.js]: No lets in Redux.

File: client/src/reducers/index.test.js [hunk 5]
```diff
@@ -1,154 +1,306 @@
...
+describe("reducer() ", () => {
 
-    it('isEmptySquare() check whether specific square is empty', () => {
-        const empty = "";
-        const notEmpty = "test";
-        console.log(isEmptySquare(empty));
-        expect(isEmptySquare(empty)).toEqual(true);
-        expect(isEmptySquare(notEmpty)).toEqual(false);
+    it("should output correct opening", () => {
+        const stateOne = {};
```
Comment (visaudi) [inline client/src/reducers/index.test.js]: turnOne? turnTwo? Or a better abstraction?

stateOne, stateTwo are vague maybe something more game specific?

**Pull request closed without merging**